godot-core/src/engine/script_instance.rs

```diff
@@ -318,32 +254,99 @@ pub fn create_script_instance<T: ScriptInstance>(rs_instance: T) -> *mut c_void
     }
 }
 
+/// Ref-guard for a mutable reference to T where T implements `ScriptInstance`.
+pub struct SiMut<'a, T: ScriptInstance> {
+    mut_ref: &'a mut T,
+    cell: Pin<&'a GdCell<T>>,
+    base_ref: &'a Base<T::Base>,
+}
```

Comment:
This should ideally have some more documentation, most notably a usage example or links to places that have one.

Also, `T` needs to be backticked :slightly_smiling_face: